crates/era-downloader/src/stream.rs

```diff
@@ -90,8 +90,31 @@ impl<Http> EraStream<Http> {
     }
 }
 
+#[derive(Debug)]
+pub struct EraMeta {
+    path: Box<Path>,
+}
+
+impl EraMeta {
+    const fn new(path: Box<Path>) -> Self {
+        Self { path }
+    }
+}
+
+impl AsRef<Path> for EraMeta {
+    fn as_ref(&self) -> &Path {
+        self.path.as_ref()
+    }
+}
+
+impl Drop for EraMeta {
+    fn drop(&mut self) {
+        let _ = std::fs::remove_file(&self.path);
```

Comment:
should we always auto remove on drop?
maybe this isn't always what we want and could be problematic,

for example, import fails, program exits, drop wipes all files?

Reply:
You're right I did have a bad feeling about this. Decided to opt-in to non-drop explicit API. Let me know your thoughts.